galata/test/jupyterlab/utils.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+import { IJupyterLabPageFixture } from '@jupyterlab/galata';
+import { ElementHandle } from '@playwright/test';
+
+const OUTER_SELECTOR = '.jp-WindowedPanel-outer';
+
+/**
+ * Measure how much of the **notebook** viewport does a cell take up.
+ */
+export async function notebookViewportRatio(
+  notebook: ElementHandle,
+  cell: ElementHandle
+): Promise<number> {
+  const scroller = (await notebook.$(
+    OUTER_SELECTOR
+  )) as ElementHandle<HTMLElement>;
+  const n = await scroller.boundingBox();
+  const c = await cell.boundingBox();
+  const cellBottom = c.y + c.height;
+  const cellTop = c.y;
+  const notebookBottom = n.y + n.height;
+  const notebookTop = n.y;
+  const visibleBottom =
+    notebookBottom > cellBottom ? cellBottom - notebookTop : 0;
+  const visibleTop = cellTop > notebookTop ? notebookBottom - cellTop : 0;
+  const intersection =
+    notebookBottom > cellBottom && cellTop > notebookTop
+      ? n.height
+      : visibleBottom + visibleTop;
+  return intersection / n.height;
+}
+/**
+ * Scroll so that the given cell is below the viewport, revealing only a given `ratio` of the cell.
+ */
+export async function scrollToAboveCell(
+  page: IJupyterLabPageFixture,
+  notebook: ElementHandle,
+  cell: ElementHandle,
+  ratio: number
```

Comment:
Yeah, naming here is not perfect, but `scrollToCell` is too generic. The intent here was to indicate that this follows the "bottom" alignment strategy; I will update it.

Reply:
Yes, naming here is not perfect, but `scrollToCell` is too generic and wrong (because this helper will for example not align to top or middle). In fact it does not scroll cell into view when ratio is zero - it scrolls it **below the viewport**. I will still update the name.

Reply:
Updated to `positionCellPartiallyBelowViewport`